integration-libs/opf/base/root/services/opf-global-message.service.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  StateWithGlobalMessage,
+  Translatable,
+} from '@spartacus/core';
+import { timer } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfGlobalMessageService extends GlobalMessageService {
+  protected isGlobalMessageDisabled = false;
+  protected disabledKeys: string[] = [];
+  protected defaultTimeout = 2000;
+  constructor(protected store: Store<StateWithGlobalMessage>) {
+    super(store);
+  }
+  /**
+   * Add one message into store
+   * @param text: string | Translatable
+   * @param type: GlobalMessageType object
+   * @param timeout: number
+   */
+  add(
+    text: string | Translatable,
+    type: GlobalMessageType,
+    timeout?: number
+  ): void {
+    if (
+      this.isGlobalMessageDisabled &&
+      this.disabledKeys?.length &&
+      (text as Translatable)?.key &&
+      this.disabledKeys.includes((text as Translatable).key as string)
+    ) {
+      return;
+    }
+    super.add(text, type, timeout);
+  }
+
+  /**
+   * disable specific keys for a period of time
+   * @param keys: string[]
+   * @param timeout: number
+   */
+  disableGlobalMessage(keys: string[], timeout?: number): void {
+    this.isGlobalMessageDisabled = true;
+    this.disabledKeys = keys;
+    timer(timeout ?? this.defaultTimeout)
+      .pipe(take(1))
+      .subscribe(() => {
+        this.isGlobalMessageDisabled = false;
+        this.disabledKeys = [];
+      });
+  }
```

Comment:
[MEDIUM]

The `opfGlobalMessageService.disableGlobalMessage()` suppresses global error messages for a hardcoded duration of 2 seconds. This approach is prone to not work correctly when the end user has a slow network connection and get a response from backend in more than 2 seconds. Instead of using a magic 2 seconds, can we use some well-determined application's event as a source of readiness and then call a (new introduced) method  `enableGlobalMessage(keys: string)`?